src/lib/security/selinux_config.rb

```diff
@@ -0,0 +1,263 @@
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+
+module Security
+  # Class for handling SELinux kernel params
+  class SelinuxConfig
+    extend Yast::I18n
+    include Yast::I18n
+    include Yast::Logger
+
+    Yast.import "Bootloader"
+    Yast.import "ProductFeatures"
+
+    # @return [Security::SelinuxConfig::Mode] the last set mode, which can be differrent to the
+    #   {#running_mode} and {#configured_mode}. A call to {#save} is needed to make it the
+    #   {#configured_mode} for the next boot.
+    attr_reader :mode
+
+    # Constructor
+    def initialize
+      @mode = initial_mode
+    end
+
+    # Returns the initial SELinux mode
+    #
+    # It can be the proposed one if running in an installation or the configured mode for a
+    # running system
+    #
+    # @return [Security::SelinuxConfig::Mode]
+    def initial_mode
+      return @initial_mode if @initial_mode
+
+      propose! if Yast::Mode.installation
+
+      @initial_mode = configured_mode
+    end
+
+    # Returns the mode applied in the running system
+    #
+    # @note the system can be booted with a different SELinux mode that the configured one. To
+    #  know the running mode getenforce tool is used.
+    #
+    # @return [Mode] running SELinux mode if command executed successfully; :disabled otherwise
+    def running_mode
+      id = Yast::Execute.locally!(GETENFORCE_PATH, stdout: :capture).chomp.downcase.to_sym
+      Mode.find(id)
+    rescue Cheetah::ExecutionFailed => e
+      log.info(e.message)
+
+      Mode.find(:disabled)
+    end
+
+    # Returns a collection holding all known SELinux modes
+    #
+    # @return [Array<Security::SelinuxConfig::Mode>] a collection of knwon SELinux modes
+    def modes
+      Mode.all
+    end
+
+    # Set the mode to given value
+    #
+    # @note using nil means to set SELinux mode as disabled.
+    #
+    # @param id [String, Symbol, nil] a SELinux mode identifier
+    # @return [Mode, nil] found SelinuxConfig::Mode by given id; nil if none
+    def mode=(id)
+      @mode = Mode.find(id)
+    end
+
+    # Set current mode options as kernel parameters for the next boot
+    #
+    # @note it does not write the changes when running in installation mode, where only sets the
+    #   kernel params in memory, since the Yast::Bootloader.Write will be performed at the end of
+    #   installation.
+    #
+    # @see Yast::Bootloader#modify_kernel_params
+    #
+    # @return [Boolean] false if mode is not set or nothing changed; true otherwise
+    def save
+      return false if mode.nil?
+
+      changed = Yast::Bootloader.modify_kernel_params(**mode.options)
+      changed = Yast::Bootloader.Write if changed && !Yast::Mode.installation
```

Comment:
you can write it as `changed &&= Yast::Bootloader.Write unless Yast::Mode.installation`